Add pstreams library

arthurl opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#sec-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

multun commented on maintainers/maintainer-list.nix:
```diff
@@ -591,6 +591,12 @@
       fingerprint = "3D2B B230 F9FA F0C5 1832  46DD 4FDC 96F1 61E7 BA8A";
     }];
   };
+  arthur = {
```
Can you move it to a separate commit ?

arthurl replied:
done.

multun commented on pkgs/development/libraries/pstreams/default.nix:
```diff
@@ -0,0 +1,41 @@
+      PStreams allows you to run another program from your C++ application and
+      to transfer data between the two programs similar to shell pipelines.
+
+      In the simplest case, a PStreams class is like a C++ wrapper for the
```
This description is way too long.
long descriptions usually look like this https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/compilers/gcc/9/default.nix#L364

c0bw3b replied:
We don't really set a maximum length [in the manual](https://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes). We have other packages with truly long `longDescription` (cf factorio or hedgewars) and it still renders properly on nixos.org package listing.

But here we can probably lose the last paragraph and still retain the essential bits of information.

arthurl replied:
Agreed. No point providing internal details here.

multun commented on pkgs/development/libraries/pstreams/default.nix:
```diff
@@ -0,0 +1,41 @@
+
+stdenv.mkDerivation rec {
+  pname = "PStreams";
+  version = "d21afef311be496e6495a5e2103eec7a53ecb817";
```
This version string is way too long. How about `d21afef` or even `1.0.1-d21afef` ?

c0bw3b replied:
Even better to stick to the stable release `1.0.1`.
The only commit that comes after doesn't seem vital and may even break programs depending on `pstreams`

arthurl replied:
Good point. The other commits apart from the last one that you've already mentioned are all testing related. Not much point moving beyond 1.0.1.

multun commented:
Would you mind fixing [the commit message format](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes)?

c0bw3b commented:
Thanks for your first contribution to nixpkgs and welcome around here @arthurl

arthurl committed:
maintainers: Add arthurl

arthurl committed:
pstreams: init at 1.0.1

arthurl commented:
> Would you mind fixing [the commit message format](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#submitting-changes)?

Fixed.

I've also added a fix to get it working on Darwin.

arthurl commented:
> Thanks for your first contribution to nixpkgs and welcome around here @arthurl

Thanks!

c0bw3b commented:
@GrahamcOfBorg build pstreams
